pkg/tcpip/transport/tcp/test/e2e/forwarder_test.go

```diff
@@ -169,6 +170,62 @@ func TestForwarderFailedConnect(t *testing.T) {
 	}
 }
 
+func TestForwarderDroppedStats(t *testing.T) {
+	// Logger that wraps t.Logf but doesn't log after the test is finished.
+	var (
+		logMu sync.Mutex
+		done  bool
+	)
+	logf := func(format string, args ...any) {
+		logMu.Lock()
+		defer logMu.Unlock()
+		if done {
+			return
+		}
+		t.Logf(format, args...)
+	}
+	defer func() {
+		logMu.Lock()
+		done = true
+		logMu.Unlock()
+	}()
+
+	const maxPayload = 100
+	const mtu = 1200
+	c := context.New(t, mtu)
+	defer c.Cleanup()
+
+	s := c.Stack()
+	const maxInFlight = 2
+	f := tcp.NewForwarder(s, 65536, maxInFlight, func(r *tcp.ForwarderRequest) {
+		var err tcpip.Error
+		c.EP, err = r.CreateEndpoint(&c.WQ)
+		if err != nil {
+			logf("CreateEndpoint err: %v", err)
+		}
+		r.Complete(false)
+	})
+	s.SetTransportProtocolHandler(tcp.ProtocolNumber, f.HandlePacket)
+
+	for i := 0; i < maxInFlight*10; i++ {
+		iss := seqnum.Value(context.TestInitialSequenceNumber + i)
+		c.SendPacket(nil, &context.Headers{
+			SrcPort: uint16(context.TestPort + i),
+			DstPort: context.StackPort,
+			Flags:   header.TCPFlagSyn,
+			SeqNum:  iss,
+			RcvWnd:  30000,
+		})
+	}
+
+	// Verify that we got some ignored packets
+	if curr := s.Stats().TCP.ForwardMaxInFlightDrop.Value(); curr == 0 {
+		t.Errorf("Expected at least one dropped connection")
+	} else {
+		t.Logf("Got %d dropped connections", curr)
```

Comment:
I would just remove the `else` and `t.Logf`, as we only really need output when the test fails.

Reply:
Removed! I also cleaned up what happens in the forwarder callback since I was getting some flaky test failures and we don't actually need to create an endpoint for this test.